cache/cache.go

```diff
@@ -44,12 +44,10 @@ func (c *Writer) SetHash(hashFunc func(lChild, rChild []byte) []byte) {
 }
 
 func (c *Writer) GetReader() (*Reader, error) {
-	err := c.validateStructure()
-	if err != nil {
+	if err := c.flush(); err != nil {
```

Comment:
Add a comment explaining how to correctly use the Reader/Writer

Reply:
I've added a detailed annotated example, explaining how to use this entire package: https://github.com/spacemeshos/merkle-tree/blob/04d583ef9076237bcbd3b94cb3a28d66eab4ac48/merkle_test.go#L437